kotlin-native/runtime/src/main/cpp/ObjCExportCoroutines.mm

```diff
@@ -48,24 +48,32 @@
   RETURN_RESULT_OF(Kotlin_ObjCExport_createContinuationArgumentImpl, completionHolder, exceptionTypes);
 }
 
+extern "C" OBJ_GETTER(Kotlin_ObjCExport_createUnitContinuationArgument, id completion, const TypeInfo** exceptionTypes) {
+  void (^typedCompletion)(NSError * _Nullable) = completion;
+  Completion twoArgumentCompletion = ^(_Nullable id result, NSError* _Nullable error) {
+    typedCompletion(error);
```

Comment:
IIRC `typedCompletion` will be retained by the containing block automatically even without ARC.

Reply:
Nice catch!
I think I can confirm this, based on the experiments and the documentation.
And it is quite important, because this code is compiled without ARC.
Could you please add your comment to the source code?

Reply:
Added a comment.

Reply:
> I'll also try to add a test checking this explicitly later.

JFTR: https://github.com/JetBrains/kotlin/compare/d8366a286182...2664a4460b41

Reply:
@SvyatoslavScherbina thanks.